action/filesystem/filesystem.go

```diff
@@ -141,3 +149,122 @@ func DirTree(root string) ([]string, error) {
 
 	return files, err
 }
+
+// LoadLastRunTime loads time of the last execution from file
+func LoadLastRunTime(pathLastRun string) (time.Time, error) {
+	// Return zero time if file doesn't exist
+	err := CheckFileExists(pathLastRun)
+	if errors.Is(err, os.ErrNotExist) {
+		return time.Time{}, nil
+	}
+
+	content, err := os.ReadFile(pathLastRun)
+	// Return zero and error on reading error
+	if err != nil {
+		slog.Warn(
+			fmt.Sprintf("Error when reading file '%s'", pathLastRun),
+			slog.Any("error", err),
+		)
+		return time.Time{}, err
+	}
+
+	// File should contain time in RFC3339Nano format
+	lastRun, err := time.Parse(time.RFC3339Nano, string(content))
+	// Return zero and error on parsing error
+	if err != nil {
+		slog.Warn(
+			fmt.Sprintf("Error when parsing time-stamp from '%s'", pathLastRun),
+			slog.Any("error", err),
+		)
+		return time.Time{}, err
+	}
+	return lastRun, nil
+}
+
+// SaveCurrentRunTime writes the current time into file
+func SaveCurrentRunTime(pathLastRun string) error {
+	// Create temporaryFilesDir
+
+	// Create directory if needed
+	dir := filepath.Dir(pathLastRun)
+	err := CheckFileExists(dir)
+	if errors.Is(err, os.ErrNotExist) {
```

Comment:
You can probably save yourself the checks here as `os.MkdirAll` will simply return `nil` if the directory already exists and do nothing in that case.

Reply:
Valid point.